### ISSUE TITLE
malloc error comes up seemingly at random

### ISSUE THREAD

**Issue by Luapulu** [mppmu/MJDSigGen.jl#5]
When using MJDSigGen, I seem to get the following error at random:
```
julia(24106,0x107048dc0) malloc: *** error for object 0x7f9a308a3400: pointer being freed was not allocated
julia(24106,0x107048dc0) malloc: *** set a breakpoint in malloc_error_break to debug

signal (6): Abort trap: 6
in expression starting at none:0
__pthread_kill at /usr/lib/system/libsystem_kernel.dylib (unknown line)
Allocations: 41926025 (Pool: 41914119; Big: 11906); GC: 36
```

I think I can reproduce the error with the following snippet but I'm unsure if the error I get there is actually the same as the error I get when I just use MJDSigGen in my other code. I don't call `signal_calc_finalize!` explicitly anywhere. I suspect julia garbage collection might be calling it though, so may be that's why I keep getting errors.
```
using MJDSigGen: signal_calc_init, signal_calc_finalize!

setup = signal_calc_init(path_to_config)
signal_calc_finalize!(setup)

# Here comes the error...
setup2 = signal_calc_init(path_to_config)
```

Edit: actually, if I don't run that snippet in Juno, but just run it in the terminal I can reproduce the error with just:
```
using MJDSigGen: signal_calc_init

setup = signal_calc_init(path_to_config)
setup2 = signal_calc_init(path_to_config)
```
This is an issue for me, since I'd like to be able to run signal generation from one config file in parallel in multiple processes.

**Comment by oschulz** [comment]
Hm, we'll have to check, the siggen code has a few global variables ... I'm not sure we ever used the wrapper with multiple configs ...

**Comment by Luapulu** [comment]
Ok, I have a workaround at least. I can run code in parallel on different processes with something like `Distributed.pmap` and everything works for me for what I want to do right now. I cannot switch configs still, so that's a bummer, but I can remove the ability to accidentally initialise two setup objects by having the module manage a global SETUP object, rather than me as the user creating and managing it.

Code example:
```
module ExampleModule

using MJDSigGen: Struct_MJD_Siggen_Setup, signal_calc_init, fieldgen

import MJDSigGen: get_signal!, outside_detector

export init_setup


SETUP = nothing

function init_setup(configpath)
    dir, f = splitdir(configpath)
    if !isdir(joinpath(dir, "fields"))
        fieldgen(configpath)
    end

    global SETUP
    if isnothing(SETUP)
        SETUP = signal_calc_init(configpath)
    else
        @warn "Attempted to reinitialise setup"
    end

    return nothing
end

"""
All functions now have methods using the global SETUP object so you need not ever specify it as a user
"""

outside_detector(location::NTuple{3, T} where T) = outside_detector(SETUP, location)

function get_signal!(pulse::DenseVector{Float32}, location::NTuple{3, T} where T)
    get_signal!(pulse, SETUP, location)
end

end
```

The code that used to cause a crash:
```
using ExampleModule

init_setup(path_to_config)
init_setup(path_to_config)
```
now just produces a warning:
```
Reading values from config file /some/path/to/configfile.config
Setup of signal calculation done
┌ Warning: Attempted to reinitialise setup
└ @ ExampleModule ~/path/to/ExampleModule/src/somefile.jl:15
```

**Comment by oschulz** [comment]
Nice workaround. Sorry, I'm a bit overloaded during the next weeks. I hope I can take a deeper look into things after, for a real fix.

**Comment by Luapulu** [comment]
Found the issue:

https://github.com/radforddc/icpc_siggen/issues/1#issue-804832874

**Comment by Luapulu** [comment]
Proposed fix

https://github.com/radforddc/icpc_siggen/pull/2#issue-570545878

**Comment by Luapulu** [comment]
Fixed by #8
